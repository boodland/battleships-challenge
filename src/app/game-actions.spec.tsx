import { fireEvent, getByRole, getByText, render } from '@testing-library/react';

import GameActions from './game-actions';

describe('GameActions', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <GameActions
        showShips={false}
        onRestartGame={() => null}
        onToggleShowShips={() => null}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should display checkbox unchecked if showShips is false', () => {
    const { container } = render(
      <GameActions
        showShips={false}
        onRestartGame={() => null}
        onToggleShowShips={() => null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should display checkbox checked if showShips is true', () => {
    const { container } = render(
      <GameActions
        showShips={true}
        onRestartGame={() => null}
        onToggleShowShips={() => null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onRestartGame when clicking the button', () => {
    const onRestartGameSpy = vi.fn();
    const { container } = render(
      <GameActions
        showShips={false}
        onRestartGame={onRestartGameSpy}
        onToggleShowShips={() => null}
      />
    );
    const button = getByRole(container, 'button');
    fireEvent.click(button);
    expect(onRestartGameSpy).toHaveBeenCalledOnce();
  });

  it('should call onToggleShowShips when clicking the input checkbox', () => {
    const onToggleShowShipsSpy = vi.fn();
    const { container } = render(
      <GameActions
        showShips={false}
        onRestartGame={() => null}
        onToggleShowShips={onToggleShowShipsSpy}
      />
    );
    const checkbox = getByRole(container, 'checkbox');
    fireEvent.click(checkbox);
    expect(onToggleShowShipsSpy).toHaveBeenCalledOnce();
  });

  it('should call onToggleShowShips when clicking the checkbox text', () => {
    const onToggleShowShipsSpy = vi.fn();
    const { container } = render(
      <GameActions
        showShips={false}
        onRestartGame={() => null}
        onToggleShowShips={onToggleShowShipsSpy}
      />
    );
    const label = getByText(container, 'Show ships');
    fireEvent.click(label);
    expect(onToggleShowShipsSpy).toHaveBeenCalledOnce();
  });
});
