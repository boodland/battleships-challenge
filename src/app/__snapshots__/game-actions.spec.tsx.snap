// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameActions > should display checkbox checked if showShips is true 1`] = `
<div>
  <div
    class="_actions_b017e8"
  >
    <button>
      Restart
    </button>
    <div>
      <input
        checked=""
        name="Show ships"
        readonly=""
        type="checkbox"
      />
      Show ships
    </div>
  </div>
</div>
`;

exports[`GameActions > should display checkbox unchecked if showShips is false 1`] = `
<div>
  <div
    class="_actions_b017e8"
  >
    <button>
      Restart
    </button>
    <div>
      <input
        name="Show ships"
        readonly=""
        type="checkbox"
      />
      Show ships
    </div>
  </div>
</div>
`;
